packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -0,0 +1,264 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'base/context.dart';
+import 'base/file_system.dart';
+import 'base/platform.dart';
+import 'build_info.dart';
+import 'globals.dart';
+
+enum Artifact {
+  chromiumDebugKeyStore,
+  classesDexJar,
+  icudtlDat,
+  libskyShellSo,
+  dartIoEntriesTxt,
+  dartVmEntryPointsTxt,
+  dartVmEntryPointsAndroidTxt,
+  genSnapshot,
+  skyShell,
+  skySnapshot,
+  snapshotDart,
+  flutterFramework,
+}
+
+String _artifactToFileName(Artifact artifact) {
+  switch (artifact) {
```

Comment:
Not 100% sure it would work, but you could try a mechanical translation of the enum name ==> a string.  Perhaps a bit easier than maintaining both the enums and this switch statement?